// myutil/util.js
var http = require("http");
var https = require("https");

module.exports = {
    /*
    options = {
        host: 'somesite.com',
        port: 443,
        path: '/some/path',
        method: 'GET',
        headers: {
            'Content-Type': 'application/json'
        }
    };*/
    exSetup: {},
    getJSON: function(options, onResult) {
        console.log("rest::getJSON");

        var prot = options.port == 443 ? https : http;
        var req = prot.request(options, function(res) {
            var output = '';
            // console.log(options.host + ':' + res.statusCode);
            res.setEncoding('utf8');

            res.on('data', function(chunk) {
                console.log("getJSON:onData called");
                output += chunk;
            });

            res.on('end', function() {
                console.log("output: " + output);
                var obj = JSON.parse(output);
                onResult(res.statusCode, obj);
            });
        });

        req.on('error', function(err) {
            console.log("getJSON:error: " + err);
            //res.send('error: ' + err.message);
        });

        req.end();
    },
    refreshRate: function() {
        var that = this;
        this.getJSON({
                host: 'openexchangerates.org',
                port: '80',
                method: 'GET',
                path: '/api/latest.json?app_id=9a76dcf49cd14aa68ea450601d81465d',
                headers: {
                    'Content-Type': 'application/json'
                }
            },
            function(statuscode, data) {
                console.log("onResult: (" + statuscode + ")" + JSON.stringify(data));

                // If not, apply to exSetup global:
                if (data) {
                    console.log("typeof exsetup:" + typeof that.exSetup);
                    that.exSetup.rates = data.rates;
                    that.exSetup.base = data.base;
                }
            });
    }
    
};
